### PR TITLE
WIP: Add deconvolutions to onnx exporter

### PR DESCRIPTION
**Pull request by KellenSunderland**
WIP for the time being, needs some cleanup.  Here for a temporary reference.

@Roshrini

### CODE REVIEW

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 1]
```diff
@@ -88,6 +88,37 @@ def parse_helper(attrs, attrs_name, alt_value=None):
             raise AttributeError("Malformed %s dimensions: %s" % (attrs_name, str(attrs_str)))
     return alt_value
 
+
+def parse_padding(attrs):
+    tuple_re = re.compile('\([0-9|,| ]+\)')
+
+    def parse_helper(attrs_name, alt_value=None):
```
Comment (Roshrini) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: there is a parse_helper function already defined. Can we reuse it?

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 2]
```diff
@@ -160,6 +191,44 @@ def convert_weights_and_inputs(node, **kwargs):
...
+
+    proc_nodes = kwargs["proc_nodes"]
+    input_node = proc_nodes[kwargs["index_lookup"][inputs[0][0]]].name
+    weights_node = proc_nodes[kwargs["index_lookup"][inputs[1][0]]].name
+
+    if num_inputs > 2:
+        bias_node = proc_nodes[kwargs["index_lookup"][inputs[2][0]]].name
+
+    attrs = node.get("attrs")
+    num_filter = int(attrs["num_filter"])
```
Comment (Roshrini) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: "num_filter" not using anywhere else. can be removed.

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 3]
```diff
@@ -160,6 +191,44 @@ def convert_weights_and_inputs(node, **kwargs):
...
+    kernel_dims = list(parse_helper(attrs, "kernel"))
+    stride_dims = list(parse_helper(attrs, "stride", [1, 1]))
+    pad_dims = parse_padding(attrs)
+    num_group = int(attrs.get("num_group", 1))
+    input_nodes = [input_node, weights_node]
+    if num_inputs > 2:
+        input_nodes.append(bias_node)
+
+    deconv_node = helper.make_node(
```
Comment (Roshrini) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: convtranspose in onnx also takes dilations, output_shape, output_padding attributes. I think we should provide those values as well.

File: python/mxnet/contrib/onnx/mx2onnx/_op_translations.py [hunk 4]
```diff
@@ -160,6 +191,44 @@ def convert_weights_and_inputs(node, **kwargs):
         return [tval_node]
 
 
+@mx_op.register("Deconvolution")
+def convert_deconvolution(node, **kwargs):
```
Comment (Roshrini) [inline python/mxnet/contrib/onnx/mx2onnx/_op_translations.py]: Can you add function description doc? pylint will throw error

**Comment by KellenSunderland** [comment]
Thanks for the comments @Roshrini .  For the time being I've merged this into the upstream project.  Hope this code is useful if you want it for reference to add deconvs.

**Pull request closed without merging**